[CHANGELOG] Prepare v5.0.0-alpha.23 (#24406)
mui-org/material-ui: CHANGELOG.md

### [Versions](https://material-ui.com/versions/)

## 5.0.0-alpha.23

###### _Jan 14, 2021_

Big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

- 📚 Only document public paths in module augmentation (#24267) @eps1lon

> - 👩‍🎤 Migrate the Paper and CssBaseline to emotion (#24397, #24176) @povilass @mnajdova - we have created an issue tracking the progress with the migration to emotion - #24405. Would be great to see the community help us with converting more components.

Reviewer: Maybe something like this?

```suggestion
- 👩‍🎤 Migrate the Paper and CssBaseline to emotion (#24397, #24176) @povilass @mnajdova

  We have reached a point where we feel confident that the new approach should make it to v5 stable. An issue has been created to track the progress with the migration to emotion: #24405. Your contribution to this effort and the ones from the community are welcome 🙌.

```

> - 📅 Various impromenets on the picker components (#24301, #24309, #24275, #24298, #24319) @m4theushw @eps1lon @huzaima @praveenkumar-kalidass

Reviewer: ```suggestion
- 📅 Various impromenets on the date picker components (#24301, #24309, #24275, #24298, #24319) @m4theushw @eps1lon @huzaima @praveenkumar-kalidass
```

### [Versions](https://material-ui.com/versions/)

## 5.0.0-alpha.23

###### _Jan 14, 2021_

Big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 📚 Support public paths in module augmentation (#24267) @eps1lon

Reviewer: ```suggestion
- 📚 Only document public paths in module augmentation (#24267) @eps1lon
```

> - 👩‍🎤 Migrate the Paper to emotion (#24397) @povilass

Reviewer: ```suggestion
- 👩‍🎤 Migrate the Paper and CssBaseline to emotion (#24397, #24176) @povilass @mnajdova 
```

Reviewer: Also likely a good opportunity to mention the migration issue and that we seek for assistance

Author: I've extended this point with seeking assistance, please take a look

> - 📚 Implemented various CI improvements (#24313) @eps1lon, (#24358) @oliviertassinari

Reviewer: We don't mention dev ops improvements unless have an obvious first order value for the developers reading the changelog highlight. I think that we should remove this line. 

```suggestion
```

> 

Reviewer: We did more work on it than usual, maybe we can mention it. 

```suggestion
- 📅 Polish the date picker components  
```

Reviewer 2: We should make it clear that these only changed internals i.e. "Refactor internals of the picker components". 

Changelogs are about what library consumers need to look out for. "Polish" is too ambigious in my opinion.

Reviewer: We have done more than improving the internals

Author: Updated to 

> - 📅 Various impromenets on the picker components (#24301, #24309, #24275, #24298, #24319) @m4theushw @eps1lon @huzaima @praveenkumar-kalidass

Reviewer 2: No, we shouldn't mix actual bugs with internal refactorings. This might make sense for a blog post but not a changelog. A changelog is a strictly technical document.

Each bug fix/feature needs its separate changelog entry.

The internal refactorings can be grouped as a slight warning: "keep an eye out for changes in behavior for these components. We did some internal changes that are not intended to change documented behavior".

Reviewer 2: Internal: #24309, #24275, #24319
Public: #24301, #24298

Author: Should we then even mention these as highlights? Or maybe just the public ones?

- And many more 🐛 bug fixes and 📚 improvements.

### `@material-ui/core@v5.0.0-alpha.23`/`@material-ui/unstyled@v5.0.0-alpha.23`

- [Container] Fix disableGutters style not applied (#24284) @povilass
- [Paper] Migrate to emotion (#24397) @povilass
- [Slider] Allow mobile VO users to interact with Sliders (#23902) @CodySchaaf
- [SwipeableDrawer] Add bleeding demo (#24268) @vicasas
- [SwipeableDrawer] Fix overflow scroll computation logic (#24225) @yann120
- [Table] Fix "more than" translation in es-ES (#24356) @vicasas

> - [Textfield] Add error color for form input with secondary color (#24290) @praveenkumar-kalidass

Reviewer: ```suggestion
- [TextField] Fix error color for form input with secondary color (#24290) @praveenkumar-kalidass
```

### `@material-ui/lab@v5.0.0-alpha.23`

- [DatePicker] Fix out of range month selection (#24301) @m4theushw
- [DatePicker] Replace withDefaultProps with useThemeProps (#24309) @m4theushw
- [DatePicker] Simplify ExtendWrapper type (#24275) @eps1lon

> - [DateRangePicker] Remove DateRangDelimiter (#24298) @huzaima

Reviewer: It's a breaking change, we should document it following the template for BCs

- [TimePicker] Add pointer cursor for clock in desktop (#24276) @praveenkumar-kalidass
- [lab] Drop usage of createStyles (#24158) @eps1lon
- [lab] Fix import paths in generated declaration files (#24380) @eps1lon

> - [lab] Prevent possible null pointer in useValidation (#24318) @eps1lon

Reviewer: This has a narrower impact, and is not about the lab package per say

```suggestion
- [DatePicker] Prevent possible null pointer in useValidation (#24318) @eps1lon
```

Reviewer 2: These are still not only targetting the DatePicker.

Reviewer: We could make them inside of [core] in this case, if you consider it's internal.

Reviewer 2: It's not devops. It's about components in the lab.

Reviewer: Ok so it's about the date pickers, at large?

Reviewer 2: > Ok so it's about the date pickers, at large? 

Date and time pickers. Not just date pickers.

> I think that we should optimize for what the developers reading the changelog would want to know. 

We agree on that

> it seems more relevant to know that it's linked to a better date picker (and all its variants). 

A time picker is not a date picker.

Reviewer: > A time picker is not a date picker.

I guess our disagreement comes from this, I was considering it close enough

Author: Should we then leave this as `lab` at this moment and merge? I'll start with the release right away

Reviewer: Maybe we could create a new label that includes `DatePicker`, `TimePicker`, `DateRangePicker`, `DateTimePicker`, Like `Pickers`

> - [lab] Reduce coupling of parsing picker input value and props (#24319) @eps1lon

Reviewer: This has a narrower impact, and is not about the lab

```suggestion
- [DatePicker] Reduce coupling of parsing picker input value and props (#24319) @eps1lon
```

Reviewer 2: It's also not about the DatePicker.

Reviewer: DatePicker has been used, so far, as a placeholder for alk the date picker components (when something more specific doesn't exist).

- [website] Improve typography theme (#24386) @oliviertassinari

### Core

- [core] Create issue labeled (#24283) @xrkffgg
- [core] Fix eslint @oliviertassinari
- [core] Skip downloading browser binaries when building docs (#24393) @eps1lon
- [core] Small changes (#24329) @oliviertassinari
- [core] Support public paths in module augmentation (#24267) @eps1lon
- [core] Update classes generation logic (#24371) @mnajdova
- [core] Update issue mark duplicate (#24311) @xrkffgg
- [core] Update issues helper version (#24379) @xrkffgg
- [test] Add pipeline task for performance monitoring (#24289) @eps1lon
- [test] Compensate for Circle CI's low performance (#24358) @oliviertassinari
- [test] Debug expensive GH actions still runing for l10nbot (#24392) @eps1lon
- [test] Move callback args to right side of assertion (#24366) @eps1lon
- [test] Persist new declaration files in CI cache (#24313) @eps1lon
- [test] Reduce download times of playwright binaries (#24364) @eps1lon
- [test] Skip expensive GitHub actions on l10nbot commits (#24303) @eps1lon
- [test] Test declaration files in TS nightly (#24391) @eps1lon
- [styles] Define useThemeProps as unstable and fix TS issues (#24383) @mnajdova

> - [styles] Fix resolution of default props (#24253) @oliviertassinari

Reviewer: 2-3 issues were open for it for the button, how about? 
```suggestion
- [Button] Fix resolution of default props (#24253) @oliviertassinari
```

And moved to the first h3
